source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -164,8 +164,41 @@ void CacheFilter::onHeaders(LookupResult&& result, Http::RequestHeaderMap& reque
     should_continue_decoding = filter_state_ == FilterState::WaitingForCacheLookup;
     filter_state_ = FilterState::NoCachedResponseFound;
     break;
-  case CacheEntryStatus::SatisfiableRange: // TODO(#10132): break response content to the ranges
-                                           // requested.
+  case CacheEntryStatus::NotSatisfiableRange:
+    lookup_result_ = std::make_unique<LookupResult>(std::move(result));
+    filter_state_ = FilterState::DecodeServingFromCache;
+    lookup_result_->headers_->setStatus(static_cast<uint64_t>(Http::Code::RangeNotSatisfiable));
+    lookup_result_->headers_->setContentLength(0);
+    lookup_result_->headers_->addCopy(Http::Headers::get().ContentRange,
+                                      absl::StrCat("bytes */", lookup_result_->content_length_));
+    // We shouldn't serve any of the body, so the response content length is 0.
+    lookup_result_->content_length_ = 0;
+    encodeCachedResponse();
+    break;
```

Comment:
@toddmgreer is this content_length_ assigning ok? Now that we're moving the entire result and then calling encodeCachedResponse, this was the best way to merge the changes.

I don't think it's confusing like it was before, but wanted to check with you.